go/vt/vttablet/tabletmanager/rpc_actions.go

```diff
@@ -94,6 +94,7 @@ func (agent *ActionAgent) changeTypeLocked(ctx context.Context, tabletType topod
 	}
 
 	// let's update our internal state (stop query service and other things)
+	// refreshTablet will runHealthCheck.
```

Comment:
Is this true? I don't see where it happens.

Reply:
I can see that `runHealthCheck` and `refreshState` do similar things with start/stop services, but it is not immediately obvious whether they will both come to the same conclusion in all situations. In the old sequence of actions, `refreshState` would leave some breadcrumbs (`disallowQueryReason`) that `runHealthCheck` took into account.
It does seem simpler and cleaner to not call `runHealthCheck` here.

Reply:
These are idempotent functions that make the system eventually converge. Many places have this call because "there's no harm". The harmful part of it is that there is no precision or clear knowledge of what state we are in and what should be done.

In this particular situation, calling the function prematurely caused a health check failure, which introduced delays that previously did not exist. I'm hoping to iterate on this to bring more clarity on what should happen when and why.

In principle, the idempotence approach is a good one, but probably not in this situation. Where I think this would be useful is synchronizing the tablet state against mysql itself.

Reply:
I'm convinced. LGTM after the comment is changed to `broadcastHealth()`.